cmd/os_retry.go

```diff
@@ -0,0 +1,95 @@
+/*
+ * Minio Cloud Storage, (C) 2018 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
```

Comment:
Is it good idea to make this as a separate package `assured` which has two functions `MkdirAll()` and `Rename()` which exactly behaving same like in `os` package but doing additional retries?  This way there won't be much code change in `minio/cmd`

Reply:
Yeah it can be done.. let me see that.

Reply:
Actually i tried and it looks too much to do, since our error messages and path length requirements are tied together. Perhaps when we move all our fs-v1-helpers.go and posix.go into a single package we could converge that.

Reply:
I will clarify more about what exactly I was trying to say.
1. `osMkdirAll()` doing minio specifics checks are not correct.  You could just do enough i.e. retry once again on `os.IsNotExist(err)`.
2. `osRename()` same (1)
3. If making as separate package is not a choice, have this functions named as `reliableMkdirAll()/reliableRename()`

Reply:
Already did the change i have kept osMkdirAll, osRename for using a single common code -- reliableMkdirAll() , reliableRename() now only looks for os.IsNotExist()